docs/MetricsSpecification.md

```diff
@@ -0,0 +1,164 @@
+  new XyzsqlDataSourceBuilder(connectionString)
+    .UseName(“My Pool”)
+    .Build();
+
+// Metrics for this connection have the name "My Pool" associated with them.
+using var connection = dataSource.OpenConnection();
+```
+
+Or, a dependency injection helper library might expose it as follows:
+
+```csharp
+builder.Services.AddXyzsqlDataSource(connectionString,
+  x => x.UseName("My Pool"));
+```
+
+If the user does not specify a pool name, the ADO.NET Provider MUST use the connection string (without a password) as the pool name in all reported metrics.
+
+The pool name MUST be associated with a metric by adding a `pool.name` tag whenever a metric is reported via an `Instrument`.
+
+## Instruments
+
+A straightforward translation of the [OpenTelemetry Instruments for Connection Pools](https://github.com/open-telemetry/semantic-conventions/blob/main/specification/metrics/semantic_conventions/database-metrics.md#connection-pools) into C# is:
+
+```csharp
+static readonly UpDownCounter<int> ConnectionsUsageCounter = Meter.CreateUpDownCounter<int>("db.client.connections.usage",
+  unit: "{connection}", description: "The number of connections that are currently in the state described by the state tag.");
+static readonly UpDownCounter<int> MaxIdleConnectionsCounter = Meter.CreateUpDownCounter<int>("db.client.connections.idle.max",
+  unit: "{connection}", description: "The maximum number of idle open connections allowed.");
+static readonly UpDownCounter<int> MinIdleConnectionsCounter = Meter.CreateUpDownCounter<int>("db.client.connections.idle.min",
+  unit: "{connection}", description: "The minimum number of idle open connections allowed.");
+static readonly UpDownCounter<int> MaxConnectionsCounter = Meter.CreateUpDownCounter<int>("db.client.connections.max",
+  unit: "{connection}", description: "The maximum number of open connections allowed.");
+static readonly UpDownCounter<int> PendingRequestsCounter = Meter.CreateUpDownCounter<int>("db.client.connections.pending_requests",
+  unit: "{request}", description: "The number of pending requests for an open connection, cumulative for the entire pool.");
+static readonly Counter<int> TimeoutsCounter = Meter.CreateCounter<int>("db.client.connections.timeouts",
+  unit: "{timeout}", description: "The number of connection timeouts that have occurred trying to obtain a connection from the pool.");
+static readonly Histogram<float> CreateTimeHistogram = Meter.CreateHistogram<float>("db.client.connections.create_time",
+  unit: "ms", description: "The time it took to create a new connection.");
+static readonly Histogram<float> WaitTimeHistogram = Meter.CreateHistogram<float>("db.client.connections.wait_time",
+  unit: "ms", description: "The time it took to obtain an open connection from the pool.");
+static readonly Histogram<float> UseTimeHistogram = Meter.CreateHistogram<float>("db.client.connections.use_time",
+  unit: "ms", description: "The time it took to create a new connection.");
```

Comment:
This is copy-paste miss, reference uses `The time between borrowing a connection and returning it to the pool.` as a description for `db.client.connections.use_time`

Reply:
Good catch; I found and fixed that copy/paste error in https://github.com/mysql-net/MySqlConnector/pull/1329/files#diff-22f5eec5ea345b8f0fd8c96f47bb2a7e59d35071c7ad2525d6f251a5ece19d5cR1963 but forgot to come back and update this example.